nixos/doc/manual/release-notes/rl-2311.section.md

```diff
@@ -306,6 +306,21 @@ The module update takes care of the new config syntax and the data itself (user
   - `keepTerminfo` controls whether `TERMINFO` and `TERMINFO_DIRS` are preserved
     for `root` and the `wheel` group.
 
+- CoreDNS can now be built with external plugins by overriding `externalPlugins` and `vendorSha256` arguments like this:
+```
+services.coredns = {
+  enable = true;
+  package = pkgs.coredns.override {
+    externalPlugins = [
+      {name = "fanout"; repo = "github.com/networkservicemesh/fanout"; version = "v1.9.1";}
+    ];
+    vendorSha256 = "<sha256 hash>";
+  };
+};
+```
+To get the necessary sha256 hash, set `vendorSha256 = lib.fakeHash;`. The build will fail and produce your sha256 hash in the error message. You should copy that hash and use it for your `vendorSha256`.
```

Comment:
Recommend `vendorHash = "";` because it's much easier and doesn't require dependency on `lib`.